test/integration/envtest/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,369 @@
+
+	Prerequisites
+		- The Gardener-Scheduler is running
+
+	BeforeSuite
+		- Parse valid Shoot from example folder and flags. Remove the Spec.SeedName.
+		- If running in TestMachinery: Scale down the GardenerController Manager
+
+	AfterSuite
+		- Delete Shoot
+        - If running in TestMachinery: Scale up the GardenerController Manager
+
+	Test: SameRegion Scheduling Strategy Test
+		1) Create Shoot in region where no Seed exists. (e.g Shoot in eu-west-1 and only Seed exists in us-east-1)
+		   Expected Output
+			 - should fail because no Seed in same region exists1)
+	Test: Minimal Distance Scheduling Strategy Test
+		1) Create Shoot in region where no Seed exists. (e.g Shoot in eu-west-1 and only Seed exists in us-east-1)
+		   Expected Output
+			 - should successfully schedule to Seed in region with minimal distance
+	Test: Api Server ShootBindingStrategy test
+		1) Request APiServer to schedule shoot to non-existing seed
+		   Expected Output
+			 - Error from ApiServer
+		2) Request APiServer to schedule shoot that is already scheduled to another seed
+		   Expected Output
+			 - Error from ApiServer
+ **/
+
+package scheduler
+
+import (
+	"context"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/scheduler/apis/config"
+	shootcontroller "github.com/gardener/gardener/pkg/scheduler/controller/shoot"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+)
+
+var _ = Describe("Scheduler tests", func() {
+
+	Context("Same Region Scheduling Strategy test", func() {
+		var (
+			ctx          = context.Background()
+			seed         *gardencorev1beta1.Seed
+			shoot        *gardencorev1beta1.Shoot
+			cloudProfile *gardencorev1beta1.CloudProfile
+			providerType = "provider-type"
+		)
+
+		BeforeEach(func() {
+			currentConfig := &config.ShootSchedulerConfiguration{ConcurrentSyncs: 1, Strategy: config.SameRegion}
+			mgr := createManager(currentConfig)
+			mgrContext, mgrCancel = context.WithCancel(ctx)
+
+			By("start manager")
+			go func() {
+				err := mgr.Start(mgrContext)
+				Expect(err).ToNot(HaveOccurred())
+			}()
+		})
+
+		AfterEach(func() {
+			patch := client.MergeFrom(shoot.DeepCopy())
+			metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, "confirmation.gardener.cloud/deletion", "true")
+			Expect(testClient.Patch(ctx, shoot, patch)).To(Succeed())
+			Expect(testClient.Delete(ctx, shoot)).To(Succeed())
+
+			Expect(testClient.Delete(ctx, cloudProfile)).To(Succeed())
+			Expect(testClient.Delete(ctx, seed)).To(Succeed())
+
+			By("Stopping Manager")
+			mgrCancel()
+		})
+
+		It("Should fail because no Seed in same region exist", func() {
```

Comment:
```suggestion
		It("should fail because no Seed in same region exist", func() {
```

(same for other `It()`s)